de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/picking/candidate/commands/CreatePickingCandidatesCommand.java

```diff
@@ -59,6 +63,8 @@ private CreatePickingCandidatesCommand(
 		this.shipmentScheduleId = request.getShipmentScheduleId();
 		this.pickFrom = request.getPickFrom();
 		this.pickingSlotId = request.getPickingSlotId();
+		final BigDecimal qty = request.getQtyToPick().toBigDecimal();
+		this.quantity = Quantity.of(qty, getShipmentScheduleUOM());
```

Comment:
* **why changing the UOM of a quantity without any UOM conversion?**

* about calling `getShipmentScheduleUOM()` - avoid calling non static methods in constructor! see for example: https://stackoverflow.com/questions/18348797/why-is-it-considered-bad-practice-to-call-a-method-from-within-a-constructor 

* in case there is a need for that information in constructor than the uom which is computed by `getShipmentScheduleUOM()` shall be fetched in constructor and shall be captured as field.
So basically the `getShipmentScheduleUOM()` should be a just a simple getter.

Reply:
Thanks for review. I agree with creating the Quantity object in the perform() method, because of the call of getShipmentScheduleUOM().
But what does it mean this "**why changing the UOM of a quantity without any UOM conversion?**"  ? So shall i do a conversion or is it ok to pass the Quantity object from request.getQtyToPick() in the perform() method?